adapters/nativery/nativery.go

```diff
@@ -0,0 +1,267 @@
+package nativery
+
+import (
+	"fmt"
+	"maps"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/metrics"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type bidReqExtNativery struct {
+	IsAMP    bool   `json:"isAmp"`
+	WidgetId string `json:"widgetId"`
+}
+
+type bidExtNativery struct {
+	BidType       string   `json:"bid_ad_media_type"`
+	BidAdvDomains []string `json:"bid_adv_domains"`
+
+	AdvertiserId  string `json:"adv_id,omitempty"`
+	BrandCategory int    `json:"brand_category_id,omitempty"`
+}
+
+type bidExt struct {
+	Nativery bidExtNativery `json:"nativery"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	reqCopy := *request
+	var errs []error
+
+	// check if the request come from AMP
+	var isAMP int
+	if reqInfo.PbsEntryPoint == metrics.ReqTypeAMP {
+		isAMP = 1
+	}
+
+	var widgetId string
+
+	// attach body request for all the impressions
+	validImps := make([]openrtb2.Imp, 0, len(request.Imp))
+	for i, imp := range request.Imp {
+		nativeryExt, err := buildNativeryExt(&imp)
+
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if i == 0 {
+			widgetId = nativeryExt.WidgetId
+		}
+
+		validImps = append(validImps, imp)
+	}
+
+	reqCopy.Imp = validImps
+	if len(reqCopy.Imp) == 0 {
+		return nil, errs
+	}
+
+	reqExt, err := getRequestExt(reqCopy.Ext)
+	if err != nil {
+		return nil, append(errs, err)
+	}
+
+	reqExtNativery, err := getNativeryExt(reqExt, isAMP, widgetId)
+	if err != nil {
+		return nil, append(errs, err)
+	}
+	adapterRequests, splitErrors := splitRequests(reqCopy.Imp, &reqCopy, reqExt, reqExtNativery, a.endpoint)
+
+	return adapterRequests, append(errs, splitErrors...)
+}
+
+func buildNativeryExt(imp *openrtb2.Imp) (openrtb_ext.ImpExtNativery, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return openrtb_ext.ImpExtNativery{}, err
```

Comment:
please add supplemental json test case where this unmarshal error condtion is hit